packages/@aws-cdk/aws-batch/lib/eks-job-definition.ts

```diff
@@ -0,0 +1,220 @@
+   * @see https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html
+   *
+   * @default - the default service account of the container
+   */
+  readonly serviceAccount?: string;
+}
+
+/**
+ * The DNS Policy for the pod used by the Job Definition
+ *
+ * @see https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy
+ */
+export enum DnsPolicy {
+  /**
+   * The Pod inherits the name resolution configuration from the node that the Pods run on
+   */
+  DEFAULT = 'Default',
+
+  /**
+   * Any DNS query that does not match the configured cluster domain suffix, such as `"www.kubernetes.io"`,
+   * is forwarded to an upstream nameserver by the DNS server.
+   * Cluster administrators may have extra stub-domain and upstream DNS servers configured.
+   */
+  CLUSTER_FIRST = 'ClusterFirst',
+
+  /**
+   * For Pods running with `hostNetwork`, you should explicitly set its DNS policy to `CLUSTER_FIRST_WITH_HOST_NET`.
+   * Otherwise, Pods running with `hostNetwork` and `CLUSTER_FIRST` will fallback to the behavior of the `DEFAULT` policy.
+   */
+  CLUSTER_FIRST_WITH_HOST_NET = 'ClusterFirstWithHostNet',
+}
+
+/**
+ * A JobDefinition that uses Eks orchestration
+ *
+ * @resource AWS::Batch::JobDefinition
+ */
+export class EksJobDefinition extends JobDefinitionBase implements IEksJobDefinition {
+  /**
+   * Import an EksJobDefinition by its arn
+   */
+  public static fromEksJobDefinitionArn(scope: Construct, id: string, eksJobDefinitionArn: string): IEksJobDefinition {
+    const stack = Stack.of(scope);
+    const jobDefinitionName = stack.splitArn(eksJobDefinitionArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
+
+    class Import extends JobDefinitionBase implements IJobDefinition {
+      public readonly jobDefinitionArn = eksJobDefinitionArn;
+      public readonly jobDefinitionName = jobDefinitionName;
+      public readonly enabled = true;
+      public readonly container = {} as any;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly container: EksContainerDefinition;
+  public readonly dnsPolicy?: DnsPolicy;
+  public readonly useHostNetwork?: boolean;
+  public readonly serviceAccount?: string;
+
+  public readonly jobDefinitionArn: string;
+  public readonly jobDefinitionName: string;
+
+  constructor(scope: Construct, id: string, props: EksJobDefinitionProps) {
+    super(scope, id, props);
+
+    this.container = props.container;
+    this.dnsPolicy = props.dnsPolicy;
+    this.useHostNetwork = props.useHostNetwork;
+    this.serviceAccount = props.serviceAccount;
+
+    const resource = new CfnJobDefinition(this, 'Resource', {
+      ...this.resourceProps,
+      type: 'container',
+      jobDefinitionName: props.jobDefinitionName,
+      eksProperties: {
+        podProperties: {
+          containers: [
+            this.container._renderContainerDefinition(),
+          ],
+          dnsPolicy: this.dnsPolicy,
+          hostNetwork: this.useHostNetwork,
```

Comment:
in the EKS interface defs, `useHostNetwork` is defaulting to `false`.  This is all a bit misleading.  Batch is defaulting `hostNetwork` to `true` (which is different than Kube defaults) because 

> Most [batch] workloads are egress-only and don't require the overhead of IP allocation for each pod for incoming connections.

The default in CDK should match Batch's default for `hostNetwork` which means `null` or not-set is true in Batch.